src/main/java/de/bonndan/nivio/model/LandscapeImpl.java

```diff
@@ -162,12 +162,13 @@ public void addGroup(Group g) {
      */
     @NonNull
     public Group getGroup(String group) {
-        if (StringUtils.isEmpty(group))
+        if (StringUtils.isEmpty(group)) {
             group = Group.COMMON;
+        }
         GroupItem groupItem = groups.get(group);
         if (groupItem == null) {
             processLog.warn("Group " + group + " was not found, returning COMMON");
-            groupItem = groups.get(Group.COMMON);
+            groupItem = new Group(Group.COMMON);
```

Comment:
`groups` can be an empty Map, so null was returned before. Is my change how it was intended to be? 

If yes, do we need that `COMMON` logic inside of `Group`?

Reply:
Has been refactored.

Reply:
Alright, but the resulting landscape still does not have a group, is this intentional?

Reply:
If you enable the `process_containsCommonGroupIfGroupWasEmpty()` in `GroupQueryResolverTest`, you should see:
````

java.util.NoSuchElementException: No value present

	at java.base/java.util.Optional.get(Optional.java:148)
	at de.bonndan.nivio.input.GroupQueryResolver.process(GroupQueryResolver.java:36)
...
````

Reply:
I added a (passing) Test that does exactly the same as the other mentioned here, but that uses a Landscape from the LandscapeFactory. 

For a safety mechanism to use the Landscape correctly: isn't it possible to enforce the same things that the LandscapeFactory does via the Landscape's constructor?

Reply:
Yes :)

Reply:
An that `get()` on the optional in `GroupQueryResolver` should have a preceding `ifPresent()` check and an `orElse()` handling :)

Reply:
I am working on it.

Reply:
I was testing with:
````
     public LandscapeImpl(@Pattern(regexp = LandscapeItem.IDENTIFIER_VALIDATION) String identifier) {
        this.identifier = identifier;
    }

    @JsonIgnore
    @Override
    public Map<String, GroupItem> getGroups() {
        if (groups.isEmpty()) {
            addGroup(new Group(Group.COMMON));
        }
        return groups;
    }
````

But then `AllGroupsLayoutTest.testWithARelation()` and `GroupResolverTest.testBlacklistOnGroupsWithRegex()` fails (aswell as tests in `OrganicLayouterTest`, but these due to the `.get()` on the Optional I guess).

So the groups map is supposed to be empty sometimes?

Reply:
Ok, I see your changes - nice 😻 

`SubLayoutTest.testWithARelation()` is failing though 😿

Reply:
Where is this code? However, I've refactoded the landscape constructor and such code is not necessary anymore.

Reply:
I didn't push it. I knew you were working on the same thing, so I was just checking that out locally.